src/frontend/src/scheduler/distributed/stage.rs

```diff
@@ -335,7 +337,8 @@ impl StageRunner {
                     task_id: id,
                 };
                 let plan_fragment = self.create_plan_fragment(id, None);
-                futures.push(self.schedule_task(task_id, plan_fragment, None));
+                let worker = self.choose_worker(&plan_fragment)?;
```

Comment:
I think it's better to put this logic in fragmenter and assign some random parallel unit id to id. This way we don't need to change scheduler.

Reply:
I think choose which worker to execute a fragment is exactly what scheduler do. In this case, dml fragments need to 
 be scheduled to source writer nodes.

Reply:
Hmm, that's also reasonable.